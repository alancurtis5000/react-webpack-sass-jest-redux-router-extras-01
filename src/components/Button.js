import React from 'react';

const click = () => {
  console.log("clicked");
}

const Button = () => {return(
  <div>
    <button onClick={click}>Push me</button>
  </div>
)}

export default Button;