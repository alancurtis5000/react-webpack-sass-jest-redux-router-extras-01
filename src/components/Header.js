import React from 'react';
import Headline from './Headline';

export const Header = () => {
  const handleOnClick = () =>{
    console.log('clicked');
    return "clicked";
  }
  return(
    <div data-test="Header"style={{width:'100%', backgroundColor:'red'}}>
      <h1> This is the Header</h1>
      <button className="click-me" data-test="click-me" onClick={handleOnClick}>Click me</button>
      <Headline header="Post" desc="Click the button to render posts"/>
    </div>
  )
} 

export default Header;