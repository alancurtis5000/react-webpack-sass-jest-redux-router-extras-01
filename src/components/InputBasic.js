import React from 'react';
import Proptypes from 'prop-types'

const InputBasic = (props) => (
  <div className="InputBasic">
    <label htmlFor={props.id}>{props.label}</label>
    <input 
      className={props.id}
      type={props.type} 
      id={props.id} 
      onChange={props.handleOnChange} 
      placeholder={props.placeholder}
      value={props.value}
    />
    {props.errors.map((error, i)=>(<div key={i} className="error" >{error}</div>))}
  </div>
)

InputBasic.defaultProps = {
  value:"",
  label:"",
  type:"text",
  id:"unique",
  placeholder:"",
  handleOnChange:(e)=>{console.log("handleOnChange", e.target.value)},
  errors:["to many characters", "Missing @ symbol"]
}

InputBasic.propTypes = {
  value: Proptypes.any,
  label: Proptypes.string,
  type: Proptypes.string,
  id: Proptypes.string,
  placeholder: Proptypes.string,
  handleOnChange: Proptypes.func,
  errors: Proptypes.array
}

export default InputBasic;