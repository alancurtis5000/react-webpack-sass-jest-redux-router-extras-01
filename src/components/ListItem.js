import React , {Component} from 'react';
import PropTypes from 'prop-types';

class ListItem extends Component{
  constructor(props){
    super(props);
  };

  render(){
    const { title, desc } = this.props;

    if(!title){
      return null;
    }
    return(
      <div data-test="ListItem" style={{display:'flex', flexDirection:'column'}} >
        <h3 data-test="title">{title}</h3>
        <div data-test="desc">{desc} </div>
      </div>
    );
  }
}

ListItem.propTypes = {
  title: PropTypes.string,
  desc: PropTypes.string
};



export default ListItem;