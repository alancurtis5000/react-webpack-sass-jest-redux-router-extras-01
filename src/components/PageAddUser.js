import React from 'react';
import { connect } from 'react-redux';
import axios from 'axios';

export const PageAddUser = (props) => {
  const handlePingServer=()=>{
    console.log("handlePingServer");
    axios.get('http://localhost:9090/api/test').then(res=>{
      console.log({res})
  })
  }

  return (
    <div className="PageAddUser" style={{display:"flex", flexDirection:"column", maxWidth:"200px", margin:"auto"}}>
      PageAddUser
      <button onClick={handlePingServer}>Ping server</button>
    </div>
  )
}

function mapStateToProps(state) {
  return {
   state
  };
};

const mapDispatchToProps = (dispatch)=>({
  // updateFakeNewUserInput: (e) => dispatch(updateFakeNewUserInput(e)),
  // startAddFakeNewUserToUsers: ( fakeNewUser ) => dispatch(startAddFakeNewUserToUsers(fakeNewUser)),
  dispatch
});

export default connect(mapStateToProps, mapDispatchToProps)(PageAddUser);