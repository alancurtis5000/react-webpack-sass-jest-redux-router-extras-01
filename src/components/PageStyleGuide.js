import React, { Component } from 'react';

class PageStyleGuide extends Component {
  render() {
    return (
      <div>
        <h1>React PageStyleGuide!</h1>
      </div>
    )
  }
}

export default PageStyleGuide;