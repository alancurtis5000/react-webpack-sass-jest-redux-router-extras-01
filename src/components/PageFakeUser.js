import React from 'react';
import { connect } from 'react-redux';
import {   } from '../actions/fakeUsers';
import { updateFakeNewUserInput , startAddFakeNewUser } from '../actions/fakeNewUser';
import map from 'lodash/map';
import InputBasic from './InputBasic';

export const PageFakeUser = (props) => {
  const renderUsers = () => {
    return map( props.fakeUsers, (user, i)=>( <div className="user" key={i}> {user.name} </div>));
  };


  const { name, email } = props.fakeNewUser
  return (
    <div className="PageFakeUser" style={{display:"flex", flexDirection:"column", maxWidth:"200px", margin:"auto"}}>
      <InputBasic label={"Name"} value={name.value} id={"name"} type={'text'} errors={name.errors} handleOnChange={props.updateFakeNewUserInput}/>
      <InputBasic label={"Email"} value={email.value} id={"email"} type={'email'} errors={email.errors} handleOnChange={props.updateFakeNewUserInput}/>
      <button onClick={()=>console.log(props)}>Log</button> 
      <button onClick={()=>props.startAddFakeNewUser(props.fakeNewUser)}>Add New User</button> 
      <div className="user-list">
        {renderUsers()}
      </div>
    </div>
  )
}

const mapDispatchToProps = {
  updateFakeNewUserInput,
  startAddFakeNewUser
};

function mapStateToProps(state) {
  return {
    fakeUsers: state.fakeUsers,
    fakeNewUser: state.fakeNewUser
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(PageFakeUser);