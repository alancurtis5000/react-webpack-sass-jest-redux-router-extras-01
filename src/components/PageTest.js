import React from 'react';
import { connect } from 'react-redux';
import { updateForm, validateForm } from '../actions/pageTest';
import map from 'lodash/map'
import InputBasic from './InputBasic';

export const PageTest = (props) => {
  const handleOnClick = () => {
    // console.log("handleOnClick");
  };
  
  const logProps = () => {
    console.log({props:props});
  }

  const handleSubmit = () =>{
    props.validateForm(props.form);
  }

  const renderInputs = () => {
    const inputs = map( props.form, (key,i)=>{
      const { label, type, id, placeholder, errors, value} = key;
      return (
        <InputBasic 
          key={i}
          label={label}
          type={type}
          id={id}
          errors={errors}
          placeholder={placeholder}
          value={value}
          handleOnChange={(e)=>props.updateForm(id, e.target.value)}
        />
      )
    })
    return inputs;
  }

  return (
    <div style={{display:"flex", justifyContent:"center", flexDirection:'column'}} >
      <h2>React PageTest!</h2>
      <div style={{display:"flex"}}>
        <button onClick={handleOnClick}>test click</button>
        <button onClick={logProps}>log props</button>
      </div>
      <div className="form" style={{display:"flex", justifyContent:"center", flexDirection:'column', width:'300px', margin:"auto"}}>
        {renderInputs()}
        <input type="text" className="tester-input"/>
        <label htmlFor="comment">Comment</label>
        <textarea className="comment" id="comment" cols="30" rows="10" maxLength="50"></textarea>
        <div className="textarea-limit">{`${50}/${50} characters`}</div>

        <div>drop down</div>
        <div> input with type ahead and drop down</div>
        <input className="submit-button" type="button" value="Submit" onClick={handleSubmit}/>
      </div>
    </div>
  )
}

const mapDispatchToProps = {
  updateForm,
  validateForm
}

function mapStateToProps(state) {
  return {
    form: state.pageTest,
    redux: state
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(PageTest);