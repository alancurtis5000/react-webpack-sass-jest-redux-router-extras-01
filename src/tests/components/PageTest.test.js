  
import React from 'react';
import { shallow , mount} from 'enzyme';
import {PageTest} from '../../components/PageTest'

let wrapper, history, validateForm;

beforeEach(()=>{
  history = { push: jest.fn() };
  validateForm = jest.fn()
  wrapper = shallow(<PageTest validateForm={validateForm}/>);
})

test('should render PageTest correctly', ()=>{
  expect(wrapper).toMatchSnapshot();
});

test('testing button click ', ()=>{
  wrapper.find('.submit-button').simulate('click');
  expect(true).toBe(true);
});


test('testing input value change ', ()=>{
  const input = wrapper.find('.tester-input')
  input.simulate('change', { target: { value: 'Hello' } })
  expect(true).toBe(true);
});