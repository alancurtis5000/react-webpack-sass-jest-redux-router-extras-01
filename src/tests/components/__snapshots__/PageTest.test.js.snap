// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PageTest correctly 1`] = `
<PageTest
  validateForm={[MockFunction]}
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <h2>
      React PageTest!
    </h2>
    <div
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <button
        onClick={[Function]}
      >
        test click
      </button>
      <button
        onClick={[Function]}
      >
        log props
      </button>
    </div>
    <div
      className="form"
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
          "margin": "auto",
          "width": "300px",
        }
      }
    >
      <input
        className="tester-input"
        type="text"
      />
      <label
        htmlFor="comment"
      >
        Comment
      </label>
      <textarea
        className="comment"
        cols="30"
        id="comment"
        maxLength="50"
        rows="10"
      />
      <div
        className="textarea-limit"
      >
        50/50 characters
      </div>
      <div>
        drop down
      </div>
      <div>
         input with type ahead and drop down
      </div>
      <input
        className="submit-button"
        onClick={[Function]}
        type="button"
        value="Submit"
      />
    </div>
  </div>
</PageTest>
`;
